errors/errors.go

```diff
@@ -0,0 +1,139 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// ErrUnknown is an unknown error type
+var ErrUnknown = New("unknown error")
```

Comment:
`const` please

Reply:
`const` wont work for error types :)

Reply:
It should be a type not an interface for constants. We can work this around but i'm thinking a better solution

Reply:
Pushed `const` changes

Reply:
i had the solution to this in the spec, as you may see

Reply:
Just noticed! awesome